pkg/compare/compare_test.go

```diff
@@ -149,12 +146,52 @@ type Test struct {
 	shouldDiffAll         bool
 	outputFormat          string
 	checks                Checks
+	patchInFile           string
+	outputPatchFile       bool
+	patchOutDir           string
+	patchOutFile          string
 }
 
 func (test *Test) getTestDir() string {
 	return path.Join(TestDirs, strings.ReplaceAll(test.name, " ", ""))
 }
 
+func (test *Test) getPatchOutFile() (string, error) {
+	if test.patchOutDir == "" {
+		patchOutDir, err := os.MkdirTemp("", "testPatches")
+		test.patchOutDir = patchOutDir
+		if err != nil {
+			return "", err // nolint:wrapcheck
+		}
+		test.patchOutFile = path.Join(path.Join(test.patchOutDir, "patches.yml"))
+	}
+	return test.patchOutFile, nil
+}
+
+func (test *Test) cleanupPatches() error {
+	return os.RemoveAll(test.patchOutDir) // nolint:wrapcheck
+}
+
+func (test *Test) Cleanup() error {
+	return test.cleanupPatches()
+}
+
+func runTest(t *testing.T, test *Test, index int, tf *cmdtesting.TestFactory, mode Mode) {
+	IOStream, _, out, _ := genericiooptions.NewTestIOStreams()
+	klog.SetOutputBySeverity("INFO", out)
+	cmd := getCommand(t, test, index, tf, &IOStream) // nolint:gosec
+	cmdutil.BehaviorOnFatal(func(str string, code int) {
+		errorStr := fmt.Sprintf("%s\nerror code:%d\n", removeInconsistentInfo(t, str), code)
+		test.checks.Err.check(t, *test, mode, errorStr)
+		panic("Expected Error Test Case")
+	})
+	defer func() {
+		_ = recover()
+		test.checks.Out.check(t, *test, mode, removeInconsistentInfo(t, out.String()))
+	}()
+	cmd.Run(cmd, []string{})
+}
+
 // TestCompareRun ensures that Run command calls the right actions
 // and returns the expected error.
 func TestCompareRun(t *testing.T) {
```

Comment:
please add a test that gets as input the patch file (that is saved in the tests directory)  and tests that the diff is supressed, and same for the save option. 
because the patch file is part of the api it should be fully tested  as a separate file 
also it will be super helpful for the review  too see examples of the patch file